app/shell-window/pages.js

```diff
@@ -203,10 +207,24 @@ export function create (opts) {
     canGoForward () { return this._canGoForward },
 
     // wrap webview loadURL to set the `loadingURL`
-    loadURL (url, opts) {
+    async loadURL (url, opts) {
+      // setting up for auto redirect
+      let redirectToDat = await beaker.browser.getSetting('auto_redirect_to_dat')
+      let hasDatAlternative = false
+      if (!url.includes('dat://')) {
+        hasDatAlternative = await DatArchive.resolveName(url).catch(err => false).then(res => !!res)
+      }
```

Comment:
This implementation looks good but `loadURL()` isn't called in every case where we need this behavior to trigger. The place it *must* run is `checkForDatAlternative()`, because that will be called reliably on every load.

Putting it in `loadURL()` will be somewhat faster because the check is done *before* we load the target. So I think it makes sense to have it in both methods.